client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -97,7 +96,13 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
 	const siteTitle = site?.name;
 	const siteDescription = site?.description;
 	const { shouldLimitGlobalStyles } = useSiteGlobalStylesStatus( site?.ID );
-	const isDesignFirstFlow = queryParams.get( 'flowToReturnTo' ) === 'design-first';
+	const isDesignFirstFlow =
+		flow === DESIGN_FIRST_FLOW || queryParams.get( 'flowToReturnTo' ) === DESIGN_FIRST_FLOW;
+
+	// The design-first flow put the checkout at the last step, so we cannot show any upsell modal.
+	// Therefore, we need to hide any feature that needs to check out right away, e.g.: Premium theme.
+	// But maybe we can enable the global styles since it's gated under the Premium plan.
+	const disableCheckoutImmediately = isDesignFirstFlow;
```

Comment:
Hmm I think this is unnecessary for this PR? As the design-first flow does not support premium plan anyway

Reply:
I just want to use another name to make it clear when people go through the codebase. When I work on this feature initially, I could not understand why we disable premium themes and global styles in the design-first flow 🥲